README
NAME
    Dist::Zilla::Plugin::Twitter - Twitter when you release with Dist::Zilla

VERSION
    version 0.015

SYNOPSIS
    In your dist.ini:

        [Twitter]
        hash_tags = #foo
        url_shortener = TinyURL

DESCRIPTION
    This plugin will use Net::Twitter to send a release notice to Twitter.
    By default, it will include a link to release on <http://metacpan.org>.

    The default configuration is as follows:

        [Twitter]
        tweet_url = https://metacpan.org/release/{{$AUTHOR_UC}}/{{$DIST}}-{{$VERSION}}/
        tweet = Released {{$DIST}}-{{$VERSION}}{{$TRIAL}} {{$URL}}
        url_shortener = TinyURL

    The "tweet_url" is shortened with WWW::Shorten::TinyURL or whichever
    other service you choose and appended to the "tweet" message. The
    following variables are available for substitution in the URL and
    message templates:

          DIST        # Foo-Bar
          MODULE      # Foo::Bar
          ABSTRACT    # Foo-Bar is a module that FooBars
          VERSION     # 1.23
          TRIAL       # -TRIAL if is_trial, empty string otherwise.
          TARBALL     # Foo-Bar-1.23.tar.gz
          AUTHOR_UC   # JOHNDOE
          AUTHOR_LC   # johndoe
          AUTHOR_PATH # J/JO/JOHNDOE
          URL         # http://tinyurl.com/...

    You must be using the "UploadToCPAN" or "FakeRelease" plugin for this
    plugin to determine your CPAN author ID.

    You can use the "hash_tags" option to append hash tags (or anything,
    really) to the end of the message generated from "tweet".

        [Twitter]
        hash_tags = #perl #cpan #foo

    after_release

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <https://metacpan.org/module/Dist::Zilla::Plugin::Twitter/>.

SOURCE
    The development version is on github at
    <http://github.com/dagolden/dist-zilla-plugin-twitter> and may be cloned
    from <git://github.com/dagolden/dist-zilla-plugin-twitter.git>

BUGS AND LIMITATIONS
    You can make new bug reports, and view existing ones, through the web
    interface at
    <https://github.com/dagolden/dist-zilla-plugin-twitter/issues>.

AUTHORS
    *   David Golden <dagolden@cpan.org>

    *   Mike Doherty <doherty@cpan.org>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2012 by David Golden.

    This is free software, licensed under:

      The Apache License, Version 2.0, January 2004



Changes
Revision history for Dist-Zilla-Plugin-Twitter

{{$NEXT}}

0.026     2014-10-02
  - Fixed the test suite for compatibility with WWW::Shorten v0.1.1,
    which removed Metamark support [gh-8]

0.025     2014-01-16
  - Twitter API is now SSL-only, so use SSL to connect to Twitter

0.024     2013-11-27
  - Add "=head1 NAME" so the pod will get munged during test

0.023     2013-10-18
  - Improved compile test

0.022     2013-09-27
  - Add new ways of interpolating metadata into tweet text, and
    use that to fix the failing tests.

0.021     2013-06-13
  - API v1.1 support [GH #5]

0.020     2013-02-02
  - Link to doherty's github repo instead of dagolden's

0.019     2013-02-02
  - Fix a typo in the default tweet spec

0.018     2013-02-02
  - Allow adding any information from META [Yanick Champoux] and
    add a link to the repo's web URL by default

0.017     2012-07-06
  - Use url_shortener=none to use the full URL (which Twitter will
    shorten with t.co)

0.016     2012-05-18
  - Don't treat the PIN as a password [GH #1]

0.015     2012-04-01 22:40:05 EST5EDT
  - Update Twitter authentication method [RT #75995 - Mike Doherty]
    - This will require all users to re-authorize the application
      at Twitter, and will fetch & store an access token + secret
      pair instead of using passwords. You may delete the section
      in your .netrc file for api.twitter.com unless you need it
      for something else.

0.014     2012-01-13 12:36:58 EST5EDT
  - distribution abstract now available for use in tweets [Harley Pig]

0.013     2011-12-01 09:06:37 EST5EDT
  - changed default URL to metacpan [Mike Doherty]

0.012     2011-11-22 17:04:03 EST5EDT
  - better handling of missing .netrc auth data [William Orr]

0.011     2011-10-07 16:11:01 EST5EDT
  - add support for -TRIAL dists [Kent Fredric]

0.010     2011-01-23 15:53:51 EST5EDT
  - add 'url_shortener' option [Mike Doherty]

0.009     2010-07-27 11:32:57 EST5EDT
  - bump Dist::Zilla prereq to 4

0.008     2010-07-26 20:14:54 EST5EDT
  - document that FakeRelease may also be used [contributed by kent fredric]
  - add some alternate sources for username since dist-zilla
    may have renamed what gets defined by release plugins
  - update dist.ini to @DAGOLDEN bundle (but still twitter it)

0.007     2010-04-29 10:09:15 EST5EDT
  - add 'hash_tags' option to append custom tags [contributed by mschout]

0.006     2010-04-28 13:10:03 EST5EDT
  - use xauth authentication

0.005     2010-04-27 10:06:14 EST5EDT
  - turn on auto collection of prerequisites (duh)

0.004     2010-04-26 17:12:38 EST5EDT
  - don't index sample distributin for testing

0.003     2010-04-26 00:26:26 EST5EDT
  - no changes -- releasing to test the new URL

0.002     2010-04-26 00:19:42 EST5EDT
  - change url from frepan to the README on cpan.cpantesters.org;
    this is one of the fastest hyperlinks to be available after upload

0.001     2010-04-25 23:46:15 EST5EDT
  - first draft


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Dist::Zilla::App::Tester" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "Dist::Zilla::Tester" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "HTTP::Response" => 0,
    "LWP::UserAgent" => 0,
    "Module::Build" => "0.3601",
    "Params::Util" => 0,
    "Sub::Exporter" => 0,
    "Test::More" => "0.88",
    "base" => 0
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Twitter when you release with Dist::Zilla",
  "dist_author" => [
    "David Golden <dagolden\@cpan.org>",
    "Mike Doherty <doherty\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Twitter",
  "dist_version" => "0.015",
  "license" => "apache",
  "module_name" => "Dist::Zilla::Plugin::Twitter",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Config::INI::Reader" => 0,
    "Config::INI::Writer" => 0,
    "Dist::Zilla" => 4,
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Util" => 0,
    "File::Spec" => 0,
    "Moose" => "0.99",
    "Net::Twitter" => 3,
    "Try::Tiny" => 0,
    "WWW::Shorten" => "3.02",
    "WWW::Shorten::Simple" => 0,
    "WWW::Shorten::TinyURL" => 0,
    "namespace::autoclean" => "0.09",
    "perl" => "5.008",
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "script_files" => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Twitter when you release with Dist::Zilla",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>, Mike Doherty <doherty\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Dist::Zilla::App::Tester" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "Dist::Zilla::Tester" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "HTTP::Response" => 0,
    "LWP::UserAgent" => 0,
    "Module::Build" => "0.3601",
    "Params::Util" => 0,
    "Sub::Exporter" => 0,
    "Test::More" => "0.88",
    "base" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Twitter",
  "EXE_FILES" => [],
  "LICENSE" => "apache",
  "NAME" => "Dist::Zilla::Plugin::Twitter",
  "PREREQ_PM" => {
    "Config::INI::Reader" => 0,
    "Config::INI::Writer" => 0,
    "Dist::Zilla" => 4,
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Util" => 0,
    "File::Spec" => 0,
    "Moose" => "0.99",
    "Net::Twitter" => 3,
    "Try::Tiny" => 0,
    "WWW::Shorten" => "3.02",
    "WWW::Shorten::Simple" => 0,
    "WWW::Shorten::TinyURL" => 0,
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.015",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



